soroban-env-host/benches/common/measure.rs

```diff
@@ -285,11 +299,49 @@ mod cpu {
     }
 }
 
-fn measure_costs_inner<HCM: HostCostMeasurement, F>(
+fn harness<HCM: HostCostMeasurement, R>(
+    host: &Host,
+    mem_tracker: &MemTracker,
+    cpu_insn_counter: &mut cpu::InstructionCounter,
+    alloc_group_token: &mut AllocationGroupToken,
+    iterations: u64,
+    runner: &mut R,
+    samples: Vec<<<HCM as HostCostMeasurement>::Runner as CostRunner>::SampleType>,
+) -> Measurement
+where
+    R: FnMut(&Host, Vec<<<HCM as HostCostMeasurement>::Runner as CostRunner>::SampleType>),
+{
+    // TODO get rid of dependency of inputs on samples
+    let sample = samples[0].clone();
```

Comment:
Why is sample being cloned here? just to feed to HCM::get_total_input below? It looks like there's been enough refactoring that get_total_input should .. actually take the whole samples array by reference -- or maybe not take anything, if all cases are actually fine delegating to the input count that was recorded by the host? Either way this seems worth cleaning up a bit.

Reply:
Hmm, now I'm unsure. It might make sense to add some commentary about the multiplicity of each argument here -- like it's not obvious to a reader of just `harness` that `samples` is all identical clones of the one initial sample, I had forgotten that while reviewing and things are split up into enough pieces now that it's hard to keep track.

Reply:
I have cleaned this part up so that the `get_total_input` is no longer dependent on the sample (only depends on the host). This has been removed.